scope_registry.go

```diff
@@ -211,3 +225,45 @@ func (r *scopeRegistry) removeWithRLock(subscopeBucket *scopeBucket, key string)
 	defer subscopeBucket.mu.Unlock()
 	delete(subscopeBucket.s, key)
 }
+
+// Records internal Metrics' cardinalities.
+func (r *scopeRegistry) reportInternalMetrics() {
+	if r.skipInternalMetrics {
+		return
+	}
+
+	counters, gauges, histograms := atomic.Int64{}, atomic.Int64{}, atomic.Int64{}
+	rootCounters, rootGauges, rootHistograms := atomic.Int64{}, atomic.Int64{}, atomic.Int64{}
+	r.ForEachScope(func(ss *scope) {
+		counterSliceLen, gaugeSliceLen, histogramSliceLen := int64(len(ss.countersSlice)), int64(len(ss.gaugesSlice)), int64(len(ss.histogramsSlice))
+		if ss.root { // Root scope is referenced across all buckets.
+			rootCounters.Store(counterSliceLen)
+			rootGauges.Store(gaugeSliceLen)
+			rootHistograms.Store(histogramSliceLen)
+			return
+		}
+		counters.Add(counterSliceLen)
+		gauges.Add(gaugeSliceLen)
+		histograms.Add(histogramSliceLen)
+	})
+
+	counters.Add(rootCounters.Load())
+	gauges.Add(rootGauges.Load())
+	histograms.Add(rootHistograms.Load())
+	log.Printf("counters: %v, gauges: %v, histograms: %v\n", counters.Load(), gauges.Load(), histograms.Load())
```

Comment:
wouldn't it be a lot of noise from this log line?

Reply:
ideally we'd want to give service owners a means to see how their application instance is doing without querying the data backend, especially since not all metrics are emitted with hostnames